realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -504,4 +514,14 @@ private int getTotalGlobalRefCount() {
 
         return totalRefCount;
     }
+
+    /**
+     * If a Realm instance is GCed but `Realm.close()` is not called before, we still want to track the cache for
+     * debugging. Adding them to the list to keep the strong ref of the cache to prevent the cache gets GCed.
+     */
+    void leak() {
+        if (!isLeaked.getAndSet(true)) {
+            leakedCaches.add(this);
```

Comment:
OK, checked the `ThreadLocal` implementation, I think the `leak()` will only be called when:
1. User doesn't call `Realm.close()` and doesn't hold any strong ref the Realm instance.
2. The thread where the Realm instances is created on is terminated.